src/backend/sgx/thread.rs

```diff
@@ -160,32 +161,96 @@ fn sgx_enarxcall<'a>(
         }
 
         item::Enarxcall {
-            num: item::enarxcall::Number::RemoveSgxPages,
-            argv: [addr, length, ..],
+            num: item::enarxcall::Number::MmapHost,
+            argv: [addr, length, prot, ..],
             ret,
             ..
         } => {
-            let mut parameters = RemovePages::new(*addr - keep.mem.addr(), *length);
-            ENCLAVE_REMOVE_PAGES
+            // Safety: The device driver checks that the request is within the
+            // shim's address range.
```

Comment:
What is with a malicious guest? Don't we want to protect against it? `addr` is basically unchecked.

Reply:
updated comment